tbot_contrib/swupdate/swupdate_script.py

```diff
@@ -0,0 +1,97 @@
+import asyncio
+import json
+import requests
+import websockets
+import argparse
+
+
+class SWUpdater:
+    "" " Python helper class for SWUpdate " ""
+
+    url_upload = "http://{}:8080/upload"
+    url_status = "ws://{}:8080/ws"
+
+    def __init__(self, path_image: str, host_name: str):
+        self.__image = path_image
+        self.__host_name = host_name
+
+    async def wait_update_finished(self, timeout: int = 300) -> None:
+        print("Wait update finished")
+
+        async def get_finish_messages() -> int:
+            async with websockets.connect(
+                self.url_status.format(self.__host_name)
+            ) as websocket:
+
+                while True:
+                    message = await websocket.recv()
+                    try:
+                        data = json.loads(message)
+                    except ValueError:
+                        data = {"type": "UNKNOWN"}
+
+                    # print(message)
+
+                    if data["type"] == "status":
+
+                        if data["status"] == "START":
+                            continue
+                        if data["status"] == "RUN":
+                            print("Updating starts")
+                            continue
+                        if data["status"] == "UNKNOWN":
+                            continue
+                        if data["status"] == "SUCCESS":
+                            print("SWUPDATE successful !")
+                            return 0
+                        if data["status"] == "FAILURE":
+                            print("SWUPDATE failed !")
+                            return 1
+                    if data["type"] == "info":
+                        print("info")
+                        continue
+
+        await asyncio.wait_for(get_finish_messages(), timeout=timeout)
+
+    def update(self, timeout: int = 300) -> int:
+        print("Start uploading image...")
+        try:
+            response = requests.post(
+                self.url_upload.format(self.__host_name),
+                files={"file": open(self.__image, "rb")},
+            )
+
+            if response.status_code != 200:
+                raise Exception(
+                    "Cannot upload software image: {}".format(response.status_code)
+                )
+
+            print(
+                "Software image uploaded successfully. Wait for installation to be finished...\n"
+            )
+            asyncio.sleep(1)
+            asyncio.get_event_loop().run_until_complete(
+                self.wait_update_finished(timeout=timeout)
+            )
+
+        except ValueError:
+            print("No connection to host, exit")
+            return 1
+
+        return 0
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("path")
+    parser.add_argument("ip")
+    args = parser.parse_args()
+
+    path = args.path
+    ip = args.ip
+
+    SWUpdater(path, ip).update()
```

Comment:
Why create two extra variables here?
```suggestion
    SWUpdater(args.path, args.ip).update()
```